lib.rs

```diff
@@ -89,14 +107,23 @@ impl<D: Decorator> Format<D> {
 
         try!(self.print_msg_header(io, &r_decorator, record));
         let mut serializer = Serializer::new(io, r_decorator);
+        let mut skip_msg_comma = NullWriter.write_fmt(record.msg()).is_ok();
```

Comment:
If I'm not wrong, this means that `msg()` is being formated twice. Once in `print_msg_header` and second time here. To save the precious cycles it would be better to make `print_msg_header` return `skip_msg_comma` value, and inside `print_msg_header` use something like `CountingWriter` that wraps another writes, and records how many bytes were written to it. Or something like that.